FrameworkSystem/DB/InstalledComponentsDB.py

```diff
@@ -0,0 +1,770 @@
+"""
+Classes and functions for easier management of the InstalledComponents database
+"""
+
+__RCSID__ = "$Id$"
+
+import datetime
+from sqlalchemy import *
+from sqlalchemy.orm import session, sessionmaker, mapper, relationship
+from sqlalchemy.orm.collections import InstrumentedList
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.sql import *
+from sqlalchemy.engine.reflection import Inspector
+from DIRAC import gConfig, gLogger, S_OK, S_ERROR
+from DIRAC.ConfigurationSystem.Client.PathFinder import getDatabaseSection
+
+metadata = MetaData()
+Base = declarative_base()
+
+class Component( Base ):
+  """
+  This class defines the schema of the Components table in the
+  InstalledComponentsDB database
+  """
+
+  __tablename__ = 'Components'
+  __table_args__ = {
+    'mysql_engine': 'InnoDB',
+    'mysql_charset': 'utf8'
+  }
+
+  componentID = Column( 'ComponentID', Integer, primary_key = True )
+  system = Column( 'System', String( 32 ), nullable = False )
+  module = Column( 'Module', String( 32 ), nullable = False )
+  cType = Column( 'Type', String( 32 ), nullable = False )
+
+  def __init__( self, system = null(), module = null(), cType = null() ):
+    self.system = system
+    self.module = module
+    self.cType = cType
+
+  def fromDict( self, dictionary ):
+    """
+    Fill the fields of the Component object from a dictionary
+    The dictionary may contain the keys: ComponentID, System, Module, Type
+    """
+
+    if dictionary.has_key( 'ComponentID' ):
```

Comment:
For these few lines (and in other places) one can do for example
```
self.system = dictionary.get( 'System', self.system )  
```
Or at least use 
```
'System' in dictionary: 
```
 (following the very last point of the coding conventions.)